perception/autoware_tensorrt_bevdet/include/autoware/tensorrt_bevdet/bevdet.hpp

```diff
@@ -0,0 +1,239 @@
+#define AUTOWARE__TENSORRT_BEVDET__BEVDET_HPP_
+
+#include "NvInfer.h"
+#include "common.hpp"
+#include "data.hpp"
+#include "postprocess.hpp"
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+
+#include <NvOnnxParser.h>
+#include <yaml-cpp/yaml.h>
+
+#include <algorithm>
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+class adjFrame
+{
+public:
+  adjFrame() {}
+  adjFrame(int _n, size_t _buf_size)
+  : n(_n), buf_size(_buf_size), scenes_token(_n), ego2global_rot(_n), ego2global_trans(_n)
+  {
+    CHECK_CUDA(cudaMalloc(reinterpret_cast<void **>(&adj_buffer), _n * _buf_size));
+    CHECK_CUDA(cudaMemset(adj_buffer, 0, _n * _buf_size));
+    last = 0;
+    buffer_num = 0;
+    init = false;
+
+    for (auto & rot : ego2global_rot) {
+      rot = Eigen::Quaternion<float>(0.f, 0.f, 0.f, 0.f);
+    }
+    for (auto & trans : ego2global_trans) {
+      trans = Eigen::Translation3f(0.f, 0.f, 0.f);
+    }
+  }
+  const std::string & lastScenesToken() const { return scenes_token[last]; }
+
+  void reset()
+  {
+    last = 0;  // origin -1
+    buffer_num = 0;
+    init = false;
+  }
+
+  void saveFrameBuffer(
+    const void * curr_buffer, const std::string & curr_token,
+    const Eigen::Quaternion<float> & _ego2global_rot,
+    const Eigen::Translation3f & _ego2global_trans)
+  {
+    int iters = init ? 1 : n;
+    while (iters--) {
+      last = (last + 1) % n;
+      CHECK_CUDA(cudaMemcpy(
+        reinterpret_cast<char *>(adj_buffer) + last * buf_size, curr_buffer, buf_size,
+        cudaMemcpyDeviceToDevice));
+      scenes_token[last] = curr_token;
+      ego2global_rot[last] = _ego2global_rot;
+      ego2global_trans[last] = _ego2global_trans;
+      buffer_num = std::min(buffer_num + 1, n);
+    }
+    init = true;
+  }
+  int havingBuffer(int idx) { return static_cast<int>(idx < buffer_num); }
+
+  const void * getFrameBuffer(int idx)
+  {
+    idx = (-idx + last + n) % n;
+    return reinterpret_cast<char *>(adj_buffer) + idx * buf_size;
+  }
+  void getEgo2Global(
+    int idx, Eigen::Quaternion<float> & adj_ego2global_rot,
+    Eigen::Translation3f & adj_ego2global_trans)
+  {
+    idx = (-idx + last + n) % n;
+    adj_ego2global_rot = ego2global_rot[idx];
+    adj_ego2global_trans = ego2global_trans[idx];
+  }
+
+  ~adjFrame() { CHECK_CUDA(cudaFree(adj_buffer)); }
+
+private:
+  int n;
+  size_t buf_size;
+
+  int last;
+  int buffer_num;
+  bool init;
+
+  std::vector<std::string> scenes_token;
+  std::vector<Eigen::Quaternion<float>> ego2global_rot;
+  std::vector<Eigen::Translation3f> ego2global_trans;
+
+  void * adj_buffer;
+};
+
+class BEVDet
+{
+public:
+  BEVDet() {}
+  BEVDet(
+    const std::string & config_file, int n_img, std::vector<Eigen::Matrix3f> _cams_intrin,
+    std::vector<Eigen::Quaternion<float>> _cams2ego_rot,
+    std::vector<Eigen::Translation3f> _cams2ego_trans, const std::string & onnx_file,
+    const std::string & engine_file);
+
+  int DoInfer(
```

Comment:
Please follow the guideline for function names: https://autowarefoundation.github.io/autoware-documentation/main/contributing/coding-guidelines/languages/cpp/#use-lower-snake-case-for-function-names-required-partially-automated

(Not only here but also applies to the whole codebase)
```suggestion
  int do_infer(
```

Reply:
Many functions in Autoware follow the camel hump naming convention.